custom_components/zidoo/remote.py

```diff
@@ -0,0 +1,269 @@
+                    state = MediaPlayerState.IDLE
+                else:
+                    self._media_info = playing_info
+                    status = playing_info.get("status")
+                    if status and status is not None:
+                        if status == 1 or status is True:
+                            state = MediaPlayerState.PLAYING
+                    mediatype = playing_info.get("source")
+                    if mediatype and mediatype is not None:
+                        if mediatype == "video":
+                            item_type = self._media_info.get("type")
+                            if item_type is not None and item_type == "tv":
+                                self._media_type = MediaType.TVSHOW
+                            else:
+                                self._media_type = MediaType.MOVIE
+                            self._source = ZCONTENT_VIDEO
+                        else:
+                            self._media_type = MediaType.MUSIC
+                            self._source = ZCONTENT_MUSIC
+                    else:
+                        self._media_type = MediaType.APP
+                    self._last_update = utcnow()
+                self._refresh_channels()
+                # debug only
+                if not state == self._last_state:
+                    _LOGGER.debug("%s New state (%s): %s", self._name, state, playing_info)
+                    self._last_state = state
+            if not state == self._last_state:
+                _LOGGER.debug("%s New state (%s)", self._name, state)
+                self._last_state = state
+
+            self._state = state
+
+        except Exception as exception_instance:  # pylint: disable=broad-except
+            _LOGGER.error(exception_instance)
+    def _refresh_channels(self):
+        if not self._source_list:
+            self._content_mapping = self._remote.load_source_list()
+            self._source_list = [ZCONTENT_VIDEO, ZCONTENT_MUSIC]
+            for key in self._content_mapping:
+                self._source_list.append(key)
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the device."""
+        return self._unique_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device info for this device."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.unique_id)},
+            manufacturer="Zidoo",
+            name=self.name,
+        )
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def current_activity(self):
+        """Return the current activity."""
+        return self._source
+
+    @property
+    def activity_list(self):
+        """List of available activities."""
+        return self._source_list
+
```

Comment:
You have the source list here.  It's not clear what the intended use for activities are.   The documentation talks about commands, but I see people are using them for lists of apps.   Since this is already covered in the MediaPlayer Entity I am thinking we should show the command list (zidoorc.KEYS).

I haven't worked out how you get a callback from the frontend UI yet.  Have you looked at this?